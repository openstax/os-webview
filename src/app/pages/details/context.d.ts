export type LocaleType = {
    locale: string;
};
export type TranslationType = Array<LocaleType & {slug: string}>;
type WebinarContent = {
    content: {
        heading: string;
        link: string;
        content: string;
    };
}
type StuffContent = {
    content: {
        id: number;
        buttonText: string | null;
        buttonUrl: string | null;
        content: string;
        contentLoggedIn: string;
        heading: string;
    };

}
type VideoContent = {
    title: string;
    description: string;
    embed: string;
}
export type ContextValues = {
    slug: string;
    translations: Array<TranslationType>;
    comingSoon: boolean;
    coverColor: string;
    meta: LocaleType;
    reverseGradient: boolean;
    title: string;
    titleImageUrl: string;
    communityResourceHeading: string;
    communityResourceLogoUrl: string;
    communityResourceUrl: string;
    communityResourceCta: string;
    communityResourceBlurb: string;
    communityResourceFeatureLinkUrl: string;
    communityResourceFeatureText: string;
    amazonIframe: string;
    webinarContent?: WebinarContent;
    freeStuffStudent: StuffContent;
    freeStuffInstructor: StuffContent;
    videos: [VideoContent[]];
    setUseCardBackground: React.Dispatch<React.SetStateAction<boolean>>;
};

export default function (): ContextValues;

export function DetailsContextProvider({
    data
}: {
    data: Omit<ContextValues, 'comingSoon'>;
}): null;
