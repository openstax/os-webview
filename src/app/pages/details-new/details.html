<template args="model">
    <div class="hero" data-slug="{model.slug}">
        <div class="content book-title">
            <h1>{model.bookTitle}</h1>
        </div>
    </div>
    <div class="content-wrapper">
        <div class="content">
            <div class="phone-view" skip="true"></div>
            <div class="bigger-view">
                <div class="tab-controller" skip="true"></div>
                <div class="tab-content" skip="true"></div>
            </div>
        </div>
    </div>
</template>
