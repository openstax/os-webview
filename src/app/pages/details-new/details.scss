@import 'pattern-library/core/pattern-library/headers';

@mixin set-color-items($color) {
    @at-root .#{$color} {
        .hero,
        .book-icon,
        .webinars .icon-cell {
            background-color: os-color($color);
            color: text-color(white);
        }

        .hero {
            background-image: linear-gradient(to right, os-color($color), lighten(os-color($color), 7));
        }

        @if ($color == yellow) {
            .book-icon,
            .webinars .icon-cell {
                color: text-color(normal);
            }
        }
    }

    @at-root .fg-#{$color} {
        .hero {
            color: os-color($color);
        }
    }
}

.details-page-v2 {

    @include wider-than($phone-max) {
        background-color: ui-color(white);

        &.card-background {
            > .content-wrapper {
                background-image: linear-gradient(to bottom, ui-color(white), ui-color(white) 10rem, ui-color(default) 25rem);

                @include wider-than($tablet-max) {
                    background-image: linear-gradient(to bottom, ui-color(white), ui-color(white) 15rem, ui-color(default) 30rem);
                }
            }
        }
    }

    .hero {
        @include set-color-items(gold);
        @include set-color-items(green);
        @include set-color-items(blue);
        @include set-color-items(deep-green);
        @include set-color-items(gray);
        @include set-color-items(light-blue);
        @include set-color-items(orange);
        @include set-color-items(yellow);
        @include set-color-items(red);

        align-items: flex-end;
        background-image: linear-gradient(75deg, #aaa, #777);
        color: text-color(normal);
        display: flex;
        min-height: 16.5rem;
        overflow: hidden;
        padding-bottom: 5rem;

        @include width-between($phone-max, $tablet-max) {
            min-height: 27.3rem;
            padding-bottom: 6rem;
        }

        @include wider-than($tablet-max) {
            min-height: 34.3rem;
            padding-bottom: 7rem;
        }

        .book-title {
            padding: 0;

            h1 {
                @include title-font(4rem);

                text-align: left;
                // white-space: nowrap;

                @include width-between($phone-max, $tablet-max) {
                    @include title-font(10rem);
                }

                @include wider-than($tablet-max) {
                    @include title-font(13rem);
                }
            }
        }
    }

    .content {
        @extend %content;

        justify-content: flex-start;
    }

    .phone-view {
        display: block;

        @include wider-than($phone-max) {
            display: none;
        }
    }

    .bigger-view {
        display: none;

        @include wider-than($phone-max) {
            display: block;
        }
    }

    .tab-controller {
        width: 100%;

        .tab {
            @include set-font(body-large);

            font-weight: normal;

            @include wider-than($tablet-max) {
                font-size: 150%;
                padding-bottom: $normal-margin;
            }
        }
    }

    .tab-content {
        padding: 4rem 0 6rem;
        width: 100%;
    }

    .unbold {
        font-weight: normal;
    }

    .resources {
        display: grid;
        grid-gap: 2 * $normal-margin  2 * $normal-margin;
        margin: 2rem 0 0;

        @include wider-than($phone-max) {
            grid-template-columns: 1fr 1fr;
        }

        @include wider-than($tablet-max) {
            grid-template-columns: 1fr 1fr 1fr;
        }

        &.ally {
            @include wider-than($phone-max) {
                grid-template-columns: 1fr 1fr 1fr;
            }

            @include wider-than($tablet-max) {
                grid-template-columns: 1fr 1fr 1fr 1fr;
            }
        }
    }

    .btn.default {
        background-color: ui-color(white);
    }

    .table-of-contents {
        list-style: none;
        margin: 0;
        padding: 0;
        padding-left: 2rem;

        > li {
            font-weight: bold;
            margin-bottom: 2rem;
        }

        ol {
            font-weight: normal;
            list-style: none;
            padding-left: 2rem;

            .box {
                font-weight: bold;
            }
        }
    }

}
