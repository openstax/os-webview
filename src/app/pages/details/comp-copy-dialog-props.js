import booksPromise from '~/models/books';
import shellBus from '~/components/shell/shell-bus';
import settings from 'settings';
import RequestForm from './request-form/request-form';

const filteredBooksPromise = booksPromise
    .then((books) => books.filter((b) => b.comp_copy_available && b.salesforce_abbreviation));

function getCompCopyDialogProps(props, userStatusPromise) {
    const dialogProps = {
        title: props.prompt
    };
    const setAltTitle = () => {
        dialogProps.htmlTitle = '<span class="fa fa-check"></span>';
        dialogProps.customClass = 'request-comp-copy-dialog';
        shellBus.emit('updateDialog');
    };
    const formHandlers = {
        done: () => shell.emit('hideDialog'),
        showConfirmation: setAltTitle
    };
    const userInfo = {};
    let salesforceTitle = '';
    let notAvailable = false;
    const formGetProps = () => Object.assign(
        {
            user: userInfo,
            salesforceTitle,
            notAvailable
        },
        props);

    dialogProps.content = new RequestForm(formGetProps, formHandlers);
    userStatusPromise.then((userStatus) => {
        Object.assign(userInfo, {
            firstName: userStatus.firstName,
            lastName: userStatus.lastName,
            email: userStatus.email
        });
        dialogProps.content.update();
    });
    filteredBooksPromise.then((books) => {
        const entry = books.find((b) => b.title === props.title);

        if (entry) {
            salesforceTitle = entry.salesforce_abbreviation;
            dialogProps.content.update();
        } else {
            notAvailable = true;
            dialogProps.content.update();
        }
    });

    return dialogProps;
}

export default getCompCopyDialogProps;
