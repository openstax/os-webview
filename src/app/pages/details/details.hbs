<style scoped>
    @import url(/app/pages/details/details.css);
    @import url(https://fonts.googleapis.com/css?family=Lato:400,900);
</style>
<div class="details-page">
        <section class="container book-info">
            <img class="book-cover" />
                <section class="details">
                    <h1 class="title"></h1>
                    <div class="tabs">
                        <a class="table-of-contents-link" class="tab"><span>Table of Contents</span></a>
                        <div class="table-of-contents hidden">
                            <div class="box"></div>
                        </div>
                        <a href="#resources" class="tab"><span>Instructor Resources</span></a>
                        <a href="#student-resources-section" class="tab"><span>Student Resources</span></a>
                        <a href="#details" class="tab"><span>Details</span></a>
                        <a href="#errata" class="tab errata-tab"><span>Errata</span></a>
                    </div>
                    <p class="blurb"></p>
                    <h2>Senior Contributing Authors</h2>
                    <div id="top-authors"></div>
                    <div class="links">
                        <a class="go-to" href="#details">See all authors</a>
                    </div>
                </section>
                <section class="floating-menu">
                    <div class="box">
                    <!-- <h3>Get this title</h3> -->
                    <div class="get-this-book"></div>
                    <div class="notes">
                        <!-- <small><sup>*</sup>Our website is Section 508 Level A accessible.</small> -->
                    </div>
                    <div class="cta">
                        <p class="interested-text">
                            Using this book, or interested in adopting it for your
                            course?
                        </p>
                        <a href="/adoption?{{title}}" class="btn">Adopt This Book</a>
                        <a href="/interest?{{title}}" class="secondary btn">get interested</a>
                    </div>
                </div>
                </section>
        </section>
        <section class="endorsement">
            <div class="container">
                <blockquote>
                    <div id="endorsement"></div>
                        <p class="attribution">
                                <span>&mdash; <span id="attribution"></span></span>
                                <span id="attribution-school"></span>
                        </p>

                </blockquote>
            </div>
            {{{strips}}}
        </section>
        <section id="resources" class="resources block">
            <div class="container">
            <div class="row">
                <h3>OpenStax Instructor Resources</h3>
                <p>
                    Access these free resources with your verified instructor account.
                    <span id="extra-instructions"></span>
                </p>
                <div id="instructor-resources" class="resource"></div>
            </div>
            <div class="row">
            <h3>Ally Resources</h3>
            <p>
                OpenStax Allies provide additional, low-cost resources for your
                book. See whatâ€™s offered for this title below and learn more about
                our <a href="/allies">Allies here</a>.</p>
            <div id='allies' class="resource"></div>
            </div>
            <div id="student-resources-section" class="row">
                <h3>Student Resources</h3>
                <p>
                    Take a look at our free student resources that are integrated
                    with your book to help you ace your course.
                </p>
                <div id="student-resources" class="resource"></div>
            </div>
            <div class="cta hidden">
                <a class="btn handbook-link">Learn how you can help bring more
                    OpenStax books to your campus!</a>
            </div>
        </div>
        </section>
            <section id="details" class="authors">
                <div class="container">
                <h2>Details</h2>
                <h3>Authors</h3>
                <div id="all-authors"></div>
            </div>
            </section>
            <section id="errata" class="errata">
                <div class="container">
                <h2>Errata</h2>
                <p>
                    All OpenStax textbooks undergo a rigorous review process. However, like
                    any professional-grade textbook, errors sometimes occur. The good part
                    is, since our books are web-based, we can make updates periodically. If
                    you have a correction to suggest, submit it here. We'll review your
                    suggestion and make necessary changes.
                </p>
                <ul class="links">
                    <li><a class="go-to errata-link">Errata list</a></li>
                    <li><a class="go-to suggest-correction-link">Suggest a correction</a></li>
                </ul>
            </div>
            </section>
</div>
