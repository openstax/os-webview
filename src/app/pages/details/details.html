<template args="model">
    <div class="hero {model.reverseGradient ? 'reverse-gradient' : ''}" data-slug="{model.slug}">
        <div class="content book-title">
            <if condition="model.titleImage">
                <h1 class="image-heading">
                    <img class="title-image" src="{model.titleImage}" alt={model.bookTitle}>
                </h1>
            </if>
        </div>
    </div>
    <div class="content-wrapper">
        <div class="content">
            <div class="phone-view" skip="true"></div>
            <div class="bigger-view">
                <div class="tab-controller" skip="true"></div>
                <div class="tab-content" skip="true"></div>
            </div>
        </div>
    </div>
</template>
