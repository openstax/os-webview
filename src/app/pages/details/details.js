import CMSPageController from '~/controllers/cms';
import $ from '~/helpers/$';
import router from '~/router';
import DetailsLoaded from './details-loaded';

export default class Details extends CMSPageController {

    static uniDescription = 'is a peer-reviewed, free, open textbook ' +
        'that covers standard scope and sequence. Access the text, authors, ' +
        'and resources here.';

    static apDescription = 'is a peer-reviewed, free, open textbook ' +
        'that adheres to Advanced PlacementÂ® frameworks. Access the full ' +
        'text here.';

    init(bookTitle) {
        this.template = () => '';
        this.view = {
            classes: ['os-loader']
        };
        bookTitle = bookTitle.toLowerCase();
        if (/^books/.test(bookTitle)) {
            this.slug = bookTitle;
        } else {
            this.slug = `books/${bookTitle}`;
        }
    }

    onDataLoaded() {
        const textTitle = $.htmlToText(this.pageData.title);

        document.querySelector('head > meta[name="description"]').content =
            `${textTitle} ${this.pageData.is_ap ? Details.apDescription : Details.uniDescription}`;
        document.title = `${textTitle} - OpenStax`;

        const model = this.pageData;
        const authors = this.pageData.book_contributing_authors;
        const senior = (author) => author.senior_author;
        const nonsenior = (author) => !author.senior_author;
        const top = (author) => author.display_at_top;

        model.allSenior = authors.filter(senior);
        model.allNonsenior = authors.filter(nonsenior);
        model.topSenior = model.allSenior.filter(top);
        model.topNonsenior = model.allNonsenior.filter(top);

        if (model.license_name) {
            model.licenseIcon = model.license_name.match(/share/i) ?
            '/images/details/by-nc-sa.svg' : '/images/details/by.svg';
        }

        model.comingSoon = this.pageData.coming_soon ? ' coming-soon' : '';

        this.regions.self.attach(new DetailsLoaded(model));
    }

    onDataError() {
        router.navigate('/404');
    }

}
