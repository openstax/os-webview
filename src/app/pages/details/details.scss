@import 'variables';
@import 'mixins';

.details-page {
    position: relative;

    h1 {
        word-wrap: break-word;

        @media (min-width: 500px) {
            margin-top: 0;
        }
    }

    h2,
    h3,
    h4 {
        letter-spacing: -0.1rem;
    }

    .row p a {
        font-weight: bolder;
    }

    .table-of-contents-link {
        cursor: pointer;
    }

    .table-of-contents {
        background-color: rgba(color(neutral-lightest), 0.97);
        bottom: 0;
        height: 100%;
        left: 0;
        opacity: 1;
        position: fixed;
        right: 0;
        top: 0;
        transition: opacity 0.3s;
        width: 100%;
        z-index: 100;

        @include desktop {
            background-color: rgba(color(neutral-dark), 0.93);
        }

        ol {
            list-style: none;
            margin: 0;
            padding: 0;

            ol {
                padding-left: 2rem;
            }
        }

        &.hidden {
            display: initial;
            height: 0;
            opacity: 0;
            overflow: hidden;
            width: 0;
        }

        .toc-remover {
            @include absolute(top 0.5rem right 0.5rem);

            cursor: pointer;
        }

        .box {
            font-weight: 500;
            max-height: 100vh;
            overflow-y: auto;
            padding: 3rem;
            position: absolute;

            @include desktop {
                background-color: rgba(color(neutral-lightest), 0.97);
                left: 50%;
                max-height: 50rem;
                max-width: 60rem;
                top: 50%;
                transform: translate3d(-50%, -50%, 0);
                width: 100%;
            }

            .subunit {
                font-weight: 300;
            }

        }
    }

    .container {
        display: flex;
        flex-flow: column nowrap;
        margin: 0 auto;
        max-width: $boxed-width;
        padding: 0 1.5rem;

        @include tablet {
            flex-flow: row wrap;
        }

        @include desktop {
            flex-flow: row wrap;
        }
    }

    .cta > .btn {
        font-variant: normal;
        padding: 1.5rem;
        text-transform: uppercase;
        width: 100%;
    }

    .btn:hover,
    .btn:focus {
        background-color: darken(color(orange), 5%);
        transform: scale3d(1, 1, 1);
    }

    .btn.hidden {
        display: none;
    }

    .book-info {
        justify-content: space-between;
        margin-bottom: 6rem;

        @media (min-width: 500px) {
            margin: 6rem auto;
        }

        .cover-wrap {
            align-self: flex-start;
            margin: 0 -1.5rem 3rem;
            order: 1;
            position: relative;
            width: 100vw;

            @media (min-width: 500px) {
                margin: 0 1.5rem 3rem 0;
                width: 25rem;
            }

            @include desktop {
                width: 25rem;
            }

            &.coming-soon::before {
                @include coming-soon;
            }

            > img {
                width: 100%;
            }
        }

        .floating-menu {
            flex: 1 auto;
            margin-bottom: 3rem;
            order: 2;

            @media (min-width: 500px) {
                flex: 0 0 auto;
                width: 20rem;
            }

            @include desktop {
                left: 50%;
                margin-bottom: 0;
                max-width: 120rem;
                order: 3;
                position: fixed;
                top: 15rem;
                transform: translate3d(-50%, 0, 0);
                width: 100%;
                z-index: 1;
            }

            &.bottom {
                bottom: 0;
                position: absolute;
                top: auto;
                z-index: 0;

                .box {
                    left: calc(100% - 26.5rem);
                    position: relative;
                    right: auto;
                    top: 0;
                }
            }

            .box {
                @include desktop {
                    background-color: rgba(color(neutral-lightest), 0.95);
                    padding: 1.5rem;
                    position: absolute;
                    right: 1.5rem;
                    top: 0;
                    width: 25rem;
                }
            }

            .notes {
                color: color(gray);
            }

            .cta {
                color: color(gray);

                .btn:last-child {
                    background-color: color(neutral-medium);
                    margin-top: 1rem;

                    &:hover,
                    &:focus {
                        background-color: darken(color(neutral-medium), 5%);
                        transform: scale3d(1, 1, 1);
                    }
                }
            }
        }

        .details {
            font-size: 1.6rem;
            line-height: 2.6rem;
            order: 3;

            @include tablet {
                flex: 1 100%;
            }

            @include desktop {
                flex: 1 0;
                order: 2;
                padding: 0 25rem 0 0;
            }

            @media (min-width: 830px) {
                padding-left: 1.5rem;
            }

            h1 {
                line-height: 1;
            }

            a {
                color: color(cyan);

                &:hover {
                    color: darken(color(cyan), 10%);
                }
            }
        }

        .tabs {
            a:not(:last-child)::after {
                color: color(neutral-dark);
                content: ' | ';
                display: inline;
            }
        }

        .links {
            margin: 1rem 0;

            a {
                position: relative;

                &::before,
                &::after {
                    border-right: 0.2rem solid;
                    content: '';
                    display: block;
                    height: 0.7rem;
                    margin-top: -0.2rem;
                    position: absolute;
                    right: -1.3rem;
                    top: 50%;
                    transform: rotate3d(0, 0, 1, 135deg);
                    width: 0;
                }

                &::after {
                    right: -1.7rem;
                    transform: rotate3d(0, 0, 1, 45deg);
                }
            }
        }
    }

    .endorsement {
        background-color: color(green);
        color: color(neutral-lightest);

        blockquote {
            position: relative;
            quotes: '\201C''\201D''\2018''\2019';

            @include desktop {
                margin-left: 7.5rem;
            }

            &::before {
                color: color(gold);
                content: open-quote;
                font-size: 15rem;
                left: -5.5rem;
                line-height: 0;
                position: absolute;
                top: 6rem;

                @include desktop {
                    font-size: 18rem;
                    left: -9rem;
                    top: 7rem;
                }
            }

            p {
                font-size: 3.2rem;
                font-style: italic;
                font-weight: 500;
                letter-spacing: -0.15rem;
                line-height: 5rem;
                margin: 0 0 3rem;

                @include phone {
                    font-size: 2.6rem;
                    letter-spacing: -0.1rem;
                    line-height: 4rem;
                }

                @include tablet {
                    font-size: 2.9rem;
                    letter-spacing: -0.1rem;
                    line-height: 4.5rem;
                }
            }

            .attribution {
                display: flex;
                flex-direction: column;
                font-size: 2.1rem;
                justify-content: flex-end;
                letter-spacing: normal;
                line-height: 2.5rem;
                margin-top: 2.5rem;
                text-align: right;

                @include phone {
                    font-size: 1.7rem;
                    line-height: 2rem;
                    margin-top: 2rem;
                }
            }
        }

        .container {
            padding: 6rem 1.5rem;

            @include desktop {
                padding: 6rem 25rem 6rem 1.5rem;
            }
        }
    }

    .strips {
        margin: 0;
    }

    .resources {
        background-color: lighten(color(neutral-lighter), 8%);
        font-size: 1.6rem;
        line-height: 2.6rem;

        h2 {
            margin-bottom: 0;
        }

        .container {
            padding: 6rem 1.5rem;

            @include desktop {
                padding: 6rem 27.5rem 6rem 1.5rem;
            }
        }

        .row {
            flex: 1 auto;
            margin-bottom: 2.5rem;
        }

        .student-resources-cta {
            background-color: color(gold);
            background-image: url('/images/details/student-resources-bg-lg.jpg');
            background-position: left 70%;
            background-repeat: no-repeat;
            background-size: cover;
            display: flex;
            min-height: 60rem;
            transition: All ease-in-out 0.3s;

            @include phone {
                min-height: 70rem;
            }

            @media screen and (min-width: 1920px) {
                background-position: right 40%;
            }

            .container {
                width: 100%;
            }

            .caption {
                align-items: flex-start;
                display: flex;
                flex-flow: column wrap;
                max-width: 56rem;
            }

            h2 {
                color: color(blue);
                font-size: 3.6rem;
                margin-bottom: 3rem;
                width: 100%;
            }

            a {
                background-color: color(yellow);
                padding: 1.5rem 3rem;

                &:hover {
                    background-color: darken(color(yellow), 5%);
                }
            }
        }

    }

    .resource {
        display: flex;
        flex-flow: column nowrap;

        @media (min-width: #{breakpoints(phone) + 1}) {
            flex-flow: row wrap;
        }

        > div {
            background-color: color(neutral-lightest);
            display: flex;
            flex-flow: column nowrap;
            justify-content: space-between;
            margin-bottom: 1.5rem;
            padding: 1.5rem;
            position: relative;

            @media (min-width: #{breakpoints(phone) + 1}) {
                flex: 0 0 48%;

                &:nth-child(odd):not(:only-child):not(:last-child) {
                    margin-right: 1.5rem;
                }

                &:last-child {
                    margin-right: 0;
                }
            }

            @media (min-width: 900px) {
                flex: 1 0 28%;
                margin-right: 1.5rem;
                max-width: 31%;
            }

            h4 {
                color: color(blue);
                margin-bottom: 1rem;
            }

            .link-and-lock {
                align-items: flex-end;
                display: flex;
                justify-content: space-between;

                a {
                    color: color(cyan);
                    font-size: 1.4rem;
                    font-weight: 500;
                    text-transform: uppercase;

                    &:hover {
                        color: darken(color(cyan), 10%);
                    }
                }

                .locked-state {
                    color: lighten(color(neutral-light), 18%);
                    font-size: 4rem;
                    margin-right: 1rem;
                }
            }
        }
    }

    .authors {
        font-size: 1.6rem;
        line-height: 2.6rem;
        padding: 6rem 0 0;

        &:last-child {
            margin-bottom: 6rem;
        }

        .container {
            flex-flow: column wrap;

            @include desktop {
                padding: 0 27.5rem 0 1.5rem;
            }
        }
    }

    .partner-info {
        .text a {
            position: relative;

            &::before,
            &::after {
                border-right: 0.2rem solid;
                content: '';
                display: block;
                height: 0.7rem;
                position: absolute;
                right: -1.3rem;
                top: 0.3rem;
                transform: rotate3d(0, 0, 1, 135deg);
                width: 0;
            }

            &::after {
                right: -1.3rem;
                top: 0.7rem;
                transform: rotate3d(0, 0, 1, 45deg);
            }
        }

        .logo {
            align-items: center;
            display: flex;
            justify-content: center;
            margin-bottom: 3rem;

            img {
                max-width: 100%;
                width: 20rem;
            }
        }
    }

    .publish-info {
        padding-top: 1rem;

        th {
            text-align: left;
            vertical-align: top;
        }

        .license {
            display: block;
            vertical-align: top;

            img {
                height: 4rem;
                padding-right: 1rem;
            }
        }
    }

    .errata {
        font-size: 1.6rem;
        line-height: 2.6rem;
        margin: 0 0 6rem;
        padding: 6rem 0 0;

        .container {
            flex-flow: column wrap;

            @include desktop {
                padding: 0 27.5rem 0 1.5rem;
            }
        }

        h2 {
            margin-bottom: 0;
        }

        .links {
            cursor: pointer;
            margin: 0;
            padding: 0;
        }

        .go-to {
            position: relative;

            &::before,
            &::after {
                border-right: 0.2rem solid;
                content: '';
                display: block;
                height: 0.7rem;
                position: absolute;
                right: -1.3rem;
                top: 0.7rem;
                transform: rotate3d(0, 0, 1, 135deg);
                width: 0;
            }

            &::after {
                right: -1.3rem;
                top: 1.1rem;
                transform: rotate3d(0, 0, 1, 45deg);
            }

            &:hover {
                color: darken(color(cyan), 10%);
            }
        }
    }
}
