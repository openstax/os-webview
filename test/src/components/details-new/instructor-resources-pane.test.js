import InstructorResourcePane from '~/pages/details-new/phone-view/instructor-resources-pane/instructor-resources-pane.js';
import details from '../../data/details.js'
const instructorResources = {
    freeResources: details.book_faculty_resources,
    paidResources: details.book_allies
};
const userInfo = {
    "id":10001697,
    "username":"staging01",
    "first_name":"Michael",
    "last_name":"Freir",
    "full_name":"Michael Freir",
    "uuid":"d2c84d89-cbb9-41c5-bb2f-a5aeecff61db",
    "support_identifier":"cs_e7186964",
    "is_test":false,
    "salesforce_contact_id":"003c000000nUzUoAAK",
    "faculty_status":"confirmed_faculty",
    "self_reported_role":"instructor",
    "self_reported_school":"Denver University",
    "contact_infos":[
        {"id":1372,
        "type":"EmailAddress",
        "value":"h1089206@mvrht.net",
        "is_verified":true,
        "is_guessed_preferred":true}
    ],
    "applications":[
        {"id":1,"name":"OpenStax Exercises"},
        {"id":5,"name":"OpenStax CMS Dev"},
        {"id":12,"name":"Openstax Payment"},
        {"id":2,"name":"OpenStax Tutor"}
    ]
};

const userStatusPromise = new Promise((resolve) => {
    resolve(userInfo);
});


describe('InstructorResourcesPane', () => {
    it('matches snapshot', () => {
        const pane = new InstructorResourcePane({
            resources: instructorResources,
            userStatusPromise: userStatusPromise
        });

        console.log('To update snapshot: node_modules/.bin/jest --updateSnapshot --testNamePattern=Resource');
        expect(pane.el.innerHTML).toMatchSnapshot();
    });
});
