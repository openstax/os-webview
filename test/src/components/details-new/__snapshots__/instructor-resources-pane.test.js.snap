// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructorResourcesPane matches snapshot 1`] = `"<h4>Free (8)</h4><div class=\\"swipe-zone\\"><div class=\\"free-resources-region\\"></div></div><h4>Paid (8)</h4><div class=\\"swipe-zone\\"><div class=\\"paid-resources-region\\"><div class=\\"logo-box\\"><a href=\\"https://www.cogbooks.com/biology/\\"><img src=\\"null\\" alt=\\"CogBooks\\"></a></div><div class=\\"logo-box\\"><a href=\\"http://theexpertta.com/biology/\\"><img src=\\"https://d3bxy9euw4e147.cloudfront.net/oscms-dev/media/original_images/ExpertTA.png\\" alt=\\"Expert TA\\"></a></div><div class=\\"logo-box\\"><a href=\\"https://marketplace.memoryscience.com/marketplace/\\"><img src=\\"https://d3bxy9euw4e147.cloudfront.net/oscms-dev/media/original_images/MemoryScience.png\\" alt=\\"Memory Science\\"></a></div><div class=\\"logo-box\\"><a href=\\"http://openstaxcollege.odigia.com/\\"><img src=\\"https://d3bxy9euw4e147.cloudfront.net/oscms-dev/media/original_images/Odigia.png\\" alt=\\"Odigia\\"></a></div><div class=\\"logo-box\\"><a href=\\"http://www2.saplinglearning.com/introductory-biology\\"><img src=\\"https://d3bxy9euw4e147.cloudfront.net/oscms-dev/media/original_images/Sapling.png\\" alt=\\"Sapling Learning\\"></a></div><div class=\\"logo-box\\"><a href=\\"http://simbio.com/\\"><img src=\\"https://d3bxy9euw4e147.cloudfront.net/oscms-dev/media/original_images/logo-SimBio.png\\" alt=\\"SimBio\\"></a></div><div class=\\"logo-box\\"><a href=\\"https://tophat.com/openstax/biology/\\"><img src=\\"https://d3bxy9euw4e147.cloudfront.net/oscms-dev/media/original_images/TopHat.png\\" alt=\\"Top Hat\\"></a></div><div class=\\"logo-box\\"><a href=\\"http://www.wiley.com/college/sc/bio/\\"><img src=\\"https://d3bxy9euw4e147.cloudfront.net/oscms-dev/media/original_images/Wiley.png\\" alt=\\"Wiley\\"></a></div></div></div>"`;
